src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -336,6 +340,15 @@ public void SniffIncomingMessage(Message message)
                 try
                 {
                     var request = (InvokeMethodRequest) message.BodyObject;
+                    if (request.Arguments != null)
```

Comment:
If CancellationSourcesExtension  can be moved into Orleans Runtime dll, can you also move that code plus RegisterCancellationToken method into CancellationSourcesExtension   static function? Just for cleaner code organization.